typed-racket-lib/typed-racket/utils/any-wrap.rkt

```diff
@@ -1,21 +1,50 @@
 #lang racket/base
 
 (require racket/match racket/contract/combinator
+         racket/class racket/unit
          racket/fixnum racket/flonum racket/extflonum
          racket/set
          racket/undefined
+         (only-in racket/udp udp?)
+         (only-in racket/future future? fsemaphore?)
+         (only-in racket/pretty pretty-print-style-table?)
          (only-in (combine-in racket/private/promise)
                   promise?
                   prop:force promise-forcer))
 
 (define (base-val? e)
   (or (number? e) (string? e) (char? e) (symbol? e)
-      (null? e) (regexp? e) (eq? undefined e) (path? e)
-      (regexp? e) (keyword? e) (bytes? e) (boolean? e) (void? e)
+      (null? e) (eq? undefined e) (path? e) (eof-object? e)
+      (regexp? e) (pregexp? e) (byte-regexp? e) (byte-pregexp? e)
+      (keyword? e) (bytes? e) (boolean? e) (void? e)
+      (continuation-prompt-tag? e) (continuation-mark-key? e)
+      (module-path? e) (resolved-module-path? e)
+      (pretty-print-style-table? e) (namespace-anchor? e)
+      (variable-reference? e) (impersonator-property? e)
+      (semaphore? e) (fsemaphore? e)
+      (bytes-converter? e)
+      (pseudo-random-generator? e)
+      (logger? e)
+      (continuation-mark-set? e)
       ;; Base values because you can only store flonums/fixnums in these
       ;; and not any higher-order values. This isn't sound if we ever
       ;; introduce bounded polymorphism for Flvector/Fxvector.
-      (flvector? e) (fxvector? e) (extflvector? e)))
+      (flvector? e) (fxvector? e) (extflvector? e) (extflonum? e)))
+
+(define (unsafe-val? e)
+  (or (syntax? e) (namespace? e) (weak-box? e) (ephemeron? e)
+      (mpair? e) (thread-cell? e) (class? e) (unit? e)
+      (compiled-module-expression? e) (compiled-expression? e)
```

Comment:
I thought I could do it by decompiling the expression, but haven't been able to yet. Marking as "safe" for now.